### PR TITLE
[ENHANCEMENT] Find in Files & Search Progress window improvements

### PR DESCRIPTION
**Pull request by pnedev**
This is a chain of several commits that correspond to [this sourceforge patch](http://sourceforge.net/p/notepad-plus/patches/659/) (slightly improved).
I have split it to make each functional modification easier to review.

I tried to re-base the [previous pull request](https://github.com/donho/notepad-plus-plus/pull/42) as it was not automatically merge-able but I don't quite know how to do it and it didn't work. That's why I created a new PR. Sorry for the inconvenience.

### CODE REVIEW

File: PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp [hunk 1]
```diff
@@ -1800,19 +1800,22 @@ void FindReplaceDlg::findAllIn(InWhat op)
 
 		_pFinder->_scintView.display();
 		_pFinder->display();
-		doSetMarkingStruct = true;
+		::SendMessage(_hParent, NPPM_DMMHIDE, 0, (LPARAM)_pFinder->getHSelf());
+		::UpdateWindow(_hParent);
+		justCreated = true;
```
Comment (ariccio) [inline PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp]: What's the point of `justCreated`?

Why not set it up at 1759, where it's first declared, *i.e.*:

```
const bool justCreated = (!_pFinder);
```

...is there some way that we create/allocate `_pFinder` but don't create it?

File: PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp [hunk 2]
```diff
@@ -357,17 +357,17 @@ int FindReplaceDlg::saveComboHistory(int id, int maxcount, vector<generic_string
 	int count = ::SendMessage(hCombo, CB_GETCOUNT, 0, 0);
 	count = min(count, maxcount);
 
-    if (count == CB_ERR) return 0;
+	if (count == CB_ERR) return 0;
 
-    if (count)
-        strings.clear();
+	if (count)
+		strings.clear();
```
Comment (ariccio) [inline PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp]: Maybe a `strings.reserve(count);` would be a good idea here?

File: PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp (hunk 1 above)
Comment (pnedev) [inline PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp]: When `FindReplaceDlg::findAllIn` is called for the first time the `_pFinder` is created. On consecutive calls to the function `_pFinder` is already there and it is not `justCreated` and the code at 1812 shouldn't be executed.

File: PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp (hunk 2 above)
Comment (pnedev) [inline PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp]: I haven't changed that. The change is caused by the automatic replacement of the leading spaces with tabs that I have done in the first patch from the chain. You can review the patches one by one to get a clearer picture of the functional changes.

**Comment by donho** [comment]
Could you redo several pull requests by spliting this one (feature by feature)?
That will make the review and merge easier. Thank you.

**Pull request closed without merging**